### PR TITLE
Create rule S6202[kotlin]: Operator "is" should be used instead of "isInstance()"

### PR DESCRIPTION
**Pull request by johann-beleites-sonarsource**

### CODE REVIEW

File: rules/S6202/kotlin/metadata.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "title": "Operator \"is\" should be used instead of \"isInstance()\""
```
[the remark this answers is not in the document]
Comment (johann-beleites-sonarsource) [reply rules/S6202/kotlin/metadata.json:2]: I'd like to also replace `A::class.java.isInstance()` and `A.javaclass.isInstance()`, hence why I didn't specify the exact `isInstance()` fun. WDYT with this in mind?

[the remark this answers is not in the document]
Comment (margarita-nedzelska-sonarsource) [reply rules/S6202/kotlin/metadata.json:2]: Maybe "Class<T>.isInstance()" will look better?

[the remark this answers is not in the document]
Comment (johann-beleites-sonarsource) [reply rules/S6202/kotlin/metadata.json:2]: Hmm what about `A::class.isInstance()`, which is refers to `KClass#isInstance()` though? :smile:

[the remark this answers is not in the document]
Comment (margarita-nedzelska-sonarsource) [reply rules/S6202/kotlin/metadata.json:2]: you're right, let's keep it "isInstance()", "Class/KClass" looks weird)

We can anyway have a more detailed message in the issue.

**Review by margarita-nedzelska-sonarsource — APPROVED** [review]

**Pull request merged**